### ISSUE TITLE
Fails with "`value` required in setHeader"

### ISSUE THREAD

**Issue by yurivict** [sharelatex/web-sharelatex#216]
web service fails right after the example document opens:
```
_http_outgoing.js:348
    throw new Error('`value` required in setHeader("' + name + '", value).');
    ^

Error: `value` required in setHeader("Host", value).
  at ClientRequest.OutgoingMessage.setHeader (_http_outgoing.js:348:11)
  at new ClientRequest (_http_client.js:80:14)
  at Object.exports.request (http.js:31:10)
  at Request.start (/usr/local/share/node-projects/sharelatex/web/node_modules/request/request.js:672:30)
  at Request.write (/usr/local/share/node-projects/sharelatex/web/node_modules/request/request.js:1314:28)
  at /usr/local/share/node-projects/sharelatex/web/node_modules/request/request.js:408:16
  at doNTCallback0 (node.js:417:9)
  at process._tickDomainCallback (node.js:387:13)
```
(with git version)

**Comment by henryoswald** [comment]
which node version are you using?

On Fri, Dec 4, 2015 at 9:04 PM, yurivict <notifications@github.com> wrote:

> web service fails right after the example document opens:
>
> _http_outgoing.js:348
>     throw new Error('`value` required in setHeader("' + name + '", value).');
>     ^
>
> Error: `value` required in setHeader("Host", value).
>   at ClientRequest.OutgoingMessage.setHeader (_http_outgoing.js:348:11)
>   at new ClientRequest (_http_client.js:80:14)
>   at Object.exports.request (http.js:31:10)
>   at Request.start (/usr/local/share/node-projects/sharelatex/web/node_modules/request/request.js:672:30)
>   at Request.write (/usr/local/share/node-projects/sharelatex/web/node_modules/request/request.js:1314:28)
>   at /usr/local/share/node-projects/sharelatex/web/node_modules/request/request.js:408:16
>   at doNTCallback0 (node.js:417:9)
>   at process._tickDomainCallback (node.js:387:13)
>
> (with git version)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sharelatex/web-sharelatex/issues/216>.
>



-- 
Henry Oswald
shareLaTeX.com co-founder

**Comment by yurivict** [comment]
Using node-4.2.1 on FreeBSD-10.2

Failed request is http://localhost:3005/user/5661390345a0265e619301d5/check

**Comment by henryoswald** [comment]
Yeah you need to downgrade. Read me should tell you details.

On Friday, 4 December 2015, yurivict <notifications@github.com> wrote:

> Using node-4.2.1 on FreeBSD-10.2
>
> Failed request is
> http://localhost:3005/user/5661390345a0265e619301d5/check
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sharelatex/web-sharelatex/issues/216#issuecomment-162107789>
> .
>


-- 
Henry Oswald
shareLaTeX.com co-founder

**Comment by yurivict** [comment]
Thanks, but it is hard to do in the context of the FreeBSD port for latex that I am creating.

**Comment by henryoswald** [comment]
Ok. For sharelatex I would look at the docker release generally.

On Friday, 4 December 2015, yurivict <notifications@github.com> wrote:

> Thanks, but it is hard to do in the context of the FreeBSD port for latex
> that I am creating.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sharelatex/web-sharelatex/issues/216#issuecomment-162111581>
> .
>


-- 
Henry Oswald
shareLaTeX.com co-founder

**Comment by yurivict** [comment]
This patch
```
--- web/node_modules/request/request.js.orig    2015-12-05 00:35:59 UTC
+++ web/node_modules/request/request.js
@@ -665,6 +665,7 @@ Request.prototype.start = function () {
 
   // We have a method named auth, which is completely different from the http.request
   // auth option.  If we don't remove it, we're gonna have a bad time.
+  self.setHeader('Host',self.uri.hostname);
   var reqOptions = copy(self)
   delete reqOptions.auth
```
fixed the problem for me with node-4.2.1. I couldn't see any other artifacts due to the node version.

So maybe you can fix this little thing and declare it node-4.2.1 compatible?

(Doker isn't an option for me - it is linux-only).

**Comment by henryoswald** [comment]
We will be releasing a node 4.2.2 version of web in the new year. There are a few changes which go into it. (there is a branch)
